File: app-miyou/src/main/java/com/miyou/app/domain/mission/port/MissionRepository.java
package com.miyou.app.domain.mission.port;

import com.miyou.app.domain.mission.model.Mission;
import com.miyou.app.domain.mission.model.MissionId;
import com.miyou.app.domain.mission.model.MissionType;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface MissionRepository {

	Mono<Mission> findById(MissionId missionId);

	Flux<Mission> findAll();

	Flux<Mission> findByType(MissionType type);

	Mono<Mission> save(Mission mission);
}

File: domain/src/main/kotlin/com/miyou/app/domain/mission/port/MissionRepository.kt
package com.miyou.app.domain.mission.port

import com.miyou.app.domain.mission.model.Mission
import com.miyou.app.domain.mission.model.MissionId
import com.miyou.app.domain.mission.model.MissionType
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface MissionRepository {
    fun findById(missionId: MissionId): Mono<Mission>

    fun findAll(): Flux<Mission>

    fun findByType(type: MissionType): Flux<Mission>

    fun save(mission: Mission): Mono<Mission>
}
